Python/_warnings.c

```diff
@@ -263,16 +257,26 @@ get_once_registry(PyInterpreterState *interp)
 
 
 static PyObject *
-get_default_action(PyInterpreterState *interp)
+get_once_registry(PyInterpreterState *interp)
 {
-    PyObject *default_action;
-
     WarningsState *st = warnings_get_state(interp);
     if (st == NULL) {
         return NULL;
     }
 
-    default_action = GET_WARNINGS_ATTR(interp, defaultaction, 0);
+    PyObject *registry;
+    Py_BEGIN_CRITICAL_SECTION(st->once_registry);
```

Comment:
We can't use the value of `st->once_registry` as the lock because that might change, so multiple callers to `get_once_registry()` may not be holding the same lock. I think it also might be `NULL`

Reply:
Right, thanks